File: boj/1924.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.StringTokenizer;

public class Main {

    private static int[] date = { 31, 28, 31, 30, 31, 30, 31, 31, 30, 31, 30, 31 };
    private static String[] day = { "SUN", "MON", "TUE", "WED", "THU", "FRI", "SAT" };

    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        StringTokenizer st = new StringTokenizer(br.readLine(), " ");
        int x = Integer.parseInt(st.nextToken());
        int y = Integer.parseInt(st.nextToken());
        int sum = 0;
        for (int i = 1; i < x; ++i) sum += date[i - 1];
        sum += y;
        System.out.println(day[sum % 7]);
    }
}

File: boj/1924.cpp
#include <iostream>
#include <vector>

using namespace std;

vector<int> date = { 31, 28, 31, 30, 31, 30, 31, 31, 30, 31, 30, 31 };
vector<string> day = { "SUN", "MON", "TUE", "WED", "THU", "FRI", "SAT" };

int main(void) {
    ios_base::sync_with_stdio(false);
    int x, y; cin >> x >> y;
    int sum = 0;
    for (int i = 1; i < x; ++i) sum += date[i - 1];
    sum += y;
    cout << day[sum % 7] << '\n';
    return 0;
}
